### ISSUE TITLE
Task 3

### ISSUE THREAD

**Issue by 123456Likhitha** [123456Likhitha/Ecommerce-website#3]
import pandas as pd

# Step 1: Load the dataset
# Replace 'your_dataset.csv' with the path to your dataset
df = pd.read_csv('your_dataset.csv')

# Step 2: Inspect the Data
print("First 5 rows of the dataset:")
print(df.head())

# Summary statistics
print("\nSummary statistics:")
print(df.describe())

# Data types of the columns
print("\nData Types of Columns:")
print(df.dtypes)

# Step 3: Handle Missing Data
# Check for missing values
missing_values = df.isnull().sum()
print("\nMissing values in each column:")
print(missing_values)

# Option 1: Drop rows with missing values
df_cleaned_drop = df.dropna()  # Drop rows with any missing values

# Option 2: Fill missing values with a specific value (e.g., mean, median, or mode)
df['column_name'] = df['column_name'].fillna(df['column_name'].mean())  # Fill with mean of the column
# Or use median or mode for numerical columns
df['column_name'] = df['column_name'].fillna(df['column_name'].median())
# For categorical columns, you could fill with mode:
df['categorical_column'] = df['categorical_column'].fillna(df['categorical_column'].mode()[0])

# Step 4: Handle Duplicates
# Check for duplicate rows
duplicates = df.duplicated().sum()
print(f"\nNumber of duplicate rows: {duplicates}")

# Remove duplicates
df = df.drop_duplicates()

# Step 5: Handle Outliers (Optional, based on specific use cases)
# One way to identify outliers is using Z-score or IQR.
# Here is a simple approach using the IQR method:

# Calculate IQR for numeric columns
Q1 = df['numeric_column'].quantile(0.25)
Q3 = df['numeric_column'].quantile(0.75)
IQR = Q3 - Q1

# Filter out values that are outside 1.5 * IQR range (lower or upper bound)
df_cleaned_outliers = df[(df['numeric_column'] >= (Q1 - 1.5 * IQR)) & (df['numeric_column'] <= (Q3 + 1.5 * IQR))]

# Step 6: Handle Incorrect Data Types
# Check if all columns have appropriate data types
print("\nCheck Data Types:")
print(df.dtypes)

# Convert data types if necessary
df['numeric_column'] = pd.to_numeric(df['numeric_column'], errors='coerce')  # Convert to numeric
df['date_column'] = pd.to_datetime(df['date_column'], errors='coerce')  # Convert to datetime

# Step 7: Handle Categorical Data (Encoding)
# For categorical columns, you may want to convert them into numeric codes or one-hot encoding

# Option 1: Convert categorical data to numeric codes
df['categorical_column'] = df['categorical_column'].astype('category').cat.codes

# Option 2: One-Hot Encoding (e.g., for 'Gender' or other categorical columns)
df = pd.get_dummies(df, columns=['categorical_column'], drop_first=True)

# Step 8: Standardization or Normalization (for certain algorithms)
# If needed, scale numeric data for machine learning algorithms that require scaled data.
from sklearn.preprocessing import StandardScaler

# Example: Standardize the 'numeric_column'
scaler = StandardScaler()
df['numeric_column'] = scaler.fit_transform(df[['numeric_column']])

# Step 9: Save the cleaned data to a new CSV file
df.to_csv('cleaned_data.csv', index=False)

# Step 10: Verify the data after cleaning
print("\nFirst 5 rows after cleaning:")
print(df.head())
